packages/pigeon/platform_tests/test_plugin/windows/pigeon/core_tests.gen.h

```diff
@@ -279,6 +284,129 @@ class AllNullableTypes {
   std::optional<AnEnum> a_nullable_enum_;
   std::optional<std::string> a_nullable_string_;
   std::optional<flutter::EncodableValue> a_nullable_object_;
+  std::optional<AllNullableTypes> all_nullable_types_;
```

Comment:
Oof. This won't work; it's the same problem as what you're fixing in Swift, basically. `std::optional` stores the actual class, not a pointer to the class, so the `AllNullableTypes` class now has infinite size.

I'll work on adjusting the C++ generators to use `std::unique_ptr` internally for any custom class storage.